Lab1_Operationalizing_Pytorch_with_Mlflow/README.md

```diff
@@ -1,31 +1,127 @@
+| 4 | Shipment and Deployment | There are many deployable assets as well as acessory tools that need specific environment and SO libraries to run. Just installing them into a machine isn't a viable option | Training  Node<br>Google Cloud Services | Docker | It's the _de facto_ pattern for shipping and deploying things |
+| 5 | Lib Management | Python has a powerful but sometimes confusing and conflicting library ecossystem. Our components may depend on conflicting libs and that will lead to problems in Production | All Nodes and Machines | Conda<br>Pip | Both are heavily used by Python community. They also work quite well with isolation provided by Docker images |
+| 6 | Remote Access | Sometimes we need to issue remote commands or operate a remote machine | All Nodes and Machines | SSH | Hey... It's SSH. There's no need to rationale :-P |
+| 7 | CI/CD & Workflow Management | We need to seamlessly integrate and deploy ML stuff. We also have to coordinate the process between nodes | All Items | Github Actions | We're using GitHub, it's heavily used by community and comes off-the-shelf |
+| 8 | Scheduler | Some activities are triggered by clock | All Nodes and Machines | CRON | We are using Unix like machines |
+| 9 | Project Metadata | We need to describe the ML Component, its structure and how it should run | Source Code Repository<br>Training Node | MLFlow Project Definition | Non-functional Requirement we must adhere |
+| 10 | ML Lifecycle Management | We must track the ML Process | All Nodes and Machines | MLFlow |  Non-functional Requirement we must adhere |
+| 11 | ML Framework | We are building ML Components. Therefore, we must use a ML framework to train our models | Training Node<br>Serving Instances | Pytorch | Non-functional Requirement we must adhere |
+
+
+### Workflow
+
+[DevOps](https://en.wikipedia.org/wiki/DevOps) tools heavily relies on actions triggered by changes
+and other events regarding application's source code. MLOps goes beyond this assumption since any reasonable
+solution must deal with assets like ML Models, datasets, features, etc. Therefore, a simples ML 
+application may turn into a complex workflow in order to ensure the best model runs in Production. The
+following workflow depicts a common development scenario where a production web application uses a ML
+Model to provide business value. 
+
+![Solution Workflow](imgs/workflow.svg)
+
+The first stage is Development Workflow. It concerns the continuous integration of new ML code and Dataset 
+changes into a Development Server. Here follows the steps:
+
+1. Commited code or changes in the Dataset fires a training execution. These events trigger Github 
+Actions (GH Actions);
+ 
+ 1. ML pipeline starts. we are considering that previous data activities (Extraction, Validation and
+ Preparation) were already executed. GH Actions issues a _gcloud commandIn to _Training Node_, so the 
```

Comment:
minor: I think it should be `_gcloud_ command to _Training Node_`. Or something like that?

Reply:
Forgot that. Fixed!